libs/sdk-core/src/swap_in/swap.rs

```diff
@@ -206,6 +211,10 @@ impl BTCReceiveSwap {
 
         // persist the swap info
         self.persister.insert_swap(swap_info.clone())?;
+
+        self.register_swap_tx_notification(chainnotifier_url, &swap_info.bitcoin_address)
+            .await?;
```

Comment:
I would put this registration in breezservices.
Keep the swapper code clean. create_swap_address means creating a swap address.
The breezservices layer is the application layer, where I think this kind of registration is most suitable.

Reply:
Sounds good to me. My main feedback was actually about the visibility of the register_swap_tx_notification which if we move the code to breezservices can be private there.